src/main/kotlin/org/rust/ide/typing/assist/RsSmartEnterProcessor.kt

```diff
@@ -36,6 +39,10 @@ class RsSmartEnterProcessor : SmartEnterProcessorWithFixers() {
             val elementType = each.node.elementType
             when {
                 elementType == LBRACE || elementType == RBRACE -> continue@loop
+                each is RsMatchArm && each.parent is RsMatchBody -> {
+                    val psiFactory = RsPsiFactory(psiFile.project)
+                    each.add(psiFactory.createComma())
```

Comment:
It's not supposed to modify PSI here
To do it, create an implementation of `SmartEnterProcessorWithFixers.Fixer` and register it in constructor. See `org.rust.ide.typing.assist.MethodCallFixer` as example